src/compiler/binder.ts

```diff
@@ -224,12 +224,24 @@ namespace ts {
 
                     // Report errors every position with duplicate declaration
                     // Report errors on previous encountered declarations
-                    let message = symbol.flags & SymbolFlags.BlockScopedVariable
+
+                    // @todo(marty) Either add a type or use the ternary operator.
+                    let message: any;
+
+                    if (symbol.name === "default") {
+                        // @todo(marty) Replace with Diagnostics.<something>
+                        message = { code: 2300, category: ts.DiagnosticCategory.Error, key: "'default' keyword used multiple times." };

```

Comment:
You need to add a diagnostic with a unique code to `diagnosticMessages.json`. Then use the appropriate generated code.